clientapi/routing/auth_fallback.go

```diff
@@ -102,6 +103,22 @@ func AuthFallback(
 	w http.ResponseWriter, req *http.Request, authType string,
 	cfg *config.ClientAPI,
 ) *util.JSONResponse {
+	// We currently only support reCaptcha, so fail early if that's not requested
+	if authType == authtypes.LoginTypeRecaptcha {
```

Comment:
Is this true? I thought we supported hCaptcha as well? But unsure it works? https://github.com/matrix-org/dendrite/issues/2878 https://github.com/matrix-org/dendrite/issues/2493

Reply:
We do, but it is also using the `authtypes.LoginTypeRecaptcha`, given it's just using a different backend for validation.